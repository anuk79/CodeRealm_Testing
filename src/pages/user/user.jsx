// @flow

// vendors
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

// components
import CandidateView from '../../components/candidate/candidate';
import LoadingSpinner from '../../components/loading-spinner/loading-spinner';

// actions
import { fetchDetails } from '../../actions/candidate.action';

const mapStateToProps = state => ({
    candidateDetails: state.candidate.candidateDetails,
    isDataFetching: state.candidate.isDataFetching,
    hasError: state.candidate.hasError
});

const mapDispatchToProps = dispatch => ({
    fetchCandidateDetails: bindActionCreators(fetchDetails, dispatch)
});
export class Candidate extends Component {
    componentDidMount() {
        this.props.fetchCandidateDetails();
    }

    render() {
        const { isDataFetching, candidateDetails, hasError } = this.props;
        let content;
        if(isDataFetching) {
            content = <LoadingSpinner customClassName="candidate-loading-spinner" />
        } else if(hasError) {
            content = (
                <div className="candidate-error">
                    {'Some error occurred. Please try again later.'}
                </div>
            );
        } else {
            content = <CandidateView candidateDetails={candidateDetails} />
        }
        return content;
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Candidate);