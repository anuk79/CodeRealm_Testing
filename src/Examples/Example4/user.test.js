
// vendors
import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// components
import { User } from './user';

describe('User component', () => {
    const mockFetchUserDetails = jest.fn();

    it('should call fetchUserDetails', () => {
        shallow(<User 
            fetching={true} 
            fetchUserDetails={mockFetchUserDetails}
        />);
        expect(mockFetchUserDetails).toBeCalled();
    });

    it('should render correctly when data is fetching', () => {
        const wrapper = shallow(<User 
            fetching={true} 
            fetchUserDetails={mockFetchUserDetails}
        />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('LoadingSpinner').length).toBe(1);
    });

    it('should render correctly when some error', () => {
        const wrapper = shallow(<User 
            errorFlag={true} 
            fetchUserDetails={mockFetchUserDetails}
        />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('.user-error').length).toBe(1);
    });

    it('should render correctly when no error and data fetching is done', () => {
        const wrapper = mount(<User 
            userDetails={{}} 
            fetchUserDetails={mockFetchUserDetails}
        />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('UserView').prop('userDetails')).toEqual({});
    });
});