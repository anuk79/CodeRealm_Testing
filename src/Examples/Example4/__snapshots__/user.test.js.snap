// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component should render correctly when data is fetching 1`] = `
<LoadingSpinner
  customClassName="user-loading-spinner"
/>
`;

exports[`User component should render correctly when no error and data fetching is done 1`] = `
<User
  fetchUserDetails={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
        Array [],
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
        Object {
          "isThrow": false,
          "value": undefined,
        },
        Object {
          "isThrow": false,
          "value": undefined,
        },
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  userDetails={Object {}}
>
  <UserView
    userDetails={Object {}}
  >
    <h3>
      User Details:
    </h3>
    <div>
      <label>
        First Name: 
      </label>
      <span />
    </div>
    <div>
      <label>
        Last Name: 
      </label>
      <span />
    </div>
  </UserView>
</User>
`;

exports[`User component should render correctly when some error 1`] = `
<div
  className="user-error"
>
  Some error occurred. Please try again later.
</div>
`;
